File: Creational Patterns/java/factoryMethod/src/main/java/factorymethod/OracleDatabase.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package factorymethod;

/**
 *
 * @author julianm
 */
public class OracleDatabase implements IDatabaseConnection{

    @Override
    public void connect() {
        System.out.println("Conectando a la base de datos Oracle...");
    }

    @Override
    public void disconnect() {
        System.out.println("Desconectando de la base de datos Oracle...");
    }
    
}

File: Creational Patterns/typeScript/factoryMethod/src/class/OracleDatabase.ts
import { IDatabaseConnection } from "../interface/IDatabaseConnection";

export class OracleDatabase implements IDatabaseConnection {
    connect(): void {
        console.log('Conectando a la base de datos Oracle...');
    }
    disconnect(): void {
        console.log('Desconectando de la base de datos Oracle...');
    }
}